ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -541,21 +687,27 @@ let find_or_create_supported_types ~__context ~pci
     Db.PCI.get_vendor_id ~__context ~self:pci
     |> Xapi_pci.int_of_id
   in
-  if vendor_id = Nvidia.nvidia_vendor_id
-  then begin
-    if is_system_display_device then []
-    else Nvidia.find_or_create_supported_types ~__context ~pci
-  end
-  else if vendor_id = Intel.intel_vendor_id
-  then
-    Intel.find_or_create_supported_types ~__context ~pci
-      ~is_system_display_device
-      ~is_host_display_enabled
-      ~is_pci_hidden
-  else begin
-    if is_system_display_device then []
-    else [find_or_create ~__context passthrough_gpu]
-  end
+  match vendor_id with
+  | x when x = Nvidia.nvidia_vendor_id -> begin
```

Comment:
Right, it's possible to match against a constant only if it's a literal, which is a bit frustrating in situations like this. I still prefer it to a whole series of `else if` though:
```
if vendor_id = Nvidia.nvidia_vendor_id then begin
    ....
end else if vendor_id = Intel.intel_vendor_id then begin
    ....
end else if vendor_id = MxGPU.amd_vendor_id then begin
    ....
end else begin
   ....
end
```